lib/jekyll/url.rb

```diff
@@ -86,7 +86,11 @@ def generate_url_from_hash(template)
 
     def generate_url_from_drop(template)
       template.gsub(%r!:([a-z_]+)!) do |match|
-        replacement = @placeholders.public_send(match.sub(":".freeze, "".freeze))
+        begin
+          replacement = @placeholders.public_send(match.sub(":".freeze, "".freeze))
+        rescue NoMethodError
```

Comment:
hmm, by using this approach, it looks like the URLDrop (at least the one i use in my tests) throws an unrelated error because it can't find `fallback_data` when using key? or `#[]`